Document Agent Builder pre-execution workflows (#7383)
elastic/docs-content: explore-analyze/ai-features/agent-builder/agents-and-workflows.md

Reviewer: ```suggestion
# Connect {{agent-builder}} agents and Elastic Workflows
```

---

> navigation_title: "Call agents from workflows"

Reviewer: ```suggestion
navigation_title: "Connect agents and workflows"
```

> navigation_title: "Call agents from workflows"

Reviewer: This page now covers both directions (calling agents from workflows and workflows that run inside an agent). The description already reflects that. I left a suggestion on the nav title and H1. :)

* For details, refer to [Set up workflows](/explore-analyze/workflows/get-started/setup.md).
* Create at least one workflow.

> ## Pre-execution workflows [pre-execution-workflows]

Reviewer: ```suggestion
## Pre-execution workflows [pre-execution-workflows]

```{applies_to}
stack: ga 9.4+
```
```

Pre-execution workflows run after each user message, before the agent makes any LLM calls in response. They let you use Elastic Workflows for deterministic preparation or control before the agent begins its reasoning loop.

:::{note}
Only administrators can configure pre-execution workflows.
:::

A pre-execution workflow runs once for each user message. It does not run before every LLM call or tool call within the agent's response.

Pre-execution workflows can:

* Add or rewrite prompt context before the agent starts.
* Cancel the agent run when a workflow detects that the request should not continue.
* Run multiple workflows in sequence when more than one workflow is assigned.

To configure pre-execution workflows:

1. Select **Manage components** at the bottom of the left sidebar, then select **Agents**.
2. Select an agent, then go to **Settings**.
3. In **Pre-execution workflow**, select **Workflows**.

> 4. Select one or more workflows to run after each user message and before the agent makes any LLM calls in response.

Reviewer: ```suggestion
3. In the **Pre-execution workflow** section, open the **Workflows** selector.
4. Select one or more workflows. They run after each user message, before the agent makes any LLM calls in response.
```
